python/test/test_publisher.py

```diff
@@ -1680,16 +1746,112 @@ def test_publish_json(self):
+                                } 
+                            }     
+                        },
+                        "formatted": {
+                            "stats": {"files": "12" + separator + "345", "errors": "1", "suites": "2", "duration": "3" + separator + "456", "tests": "4", "tests_succ": "5",
+                                      "tests_skip": "6", "tests_fail": "7", "tests_error": "8" + separator + "901", "runs": "9", "runs_succ": "10",
+                                      "runs_skip": "11", "runs_fail": "12", "runs_error": "1" + separator + "345", "commit": "commit"},
+                            "stats_with_delta": {"files": {"number": "1" + separator + "234", "delta": "-1" + separator + "234"}, "errors": "2",
+                                                 "suites": {"number": "2", "delta": "-2"}, "duration": {"number": "3" + separator + "456", "delta": "-3" + separator + "456"},
+                                                 "tests": {"number": "4", "delta": "-4"}, "tests_succ": {"number": "5", "delta": "-5"},
+                                                 "tests_skip": {"number": "6", "delta": "-6"}, "tests_fail": {"number": "7", "delta": "-7"},
+                                                 "tests_error": {"number": "8", "delta": "-8"}, "runs": {"number": "9", "delta": "-9"},
+                                                 "runs_succ": {"number": "10", "delta": "-10"},
+                                                 "runs_skip": {"number": "11", "delta": "-11"},
+                                                 "runs_fail": {"number": "12", "delta": "-12"},
+                                                 "runs_error": {"number": "1" + separator + "345", "delta": "-1" + separator + "345"}, "commit": "commit",
+                                                 "reference_type": "type", "reference_commit": "ref"}
+                        }
+                    }
+                    gha.add_to_output.assert_called_once_with('json', json.dumps(expected, ensure_ascii=False))
+
+    
+
+    def test_publish_json_with_test_cases(self):
+        for separator in ['.', ',', ' ', punctuation_space]:
+            with self.subTest(json_thousands_separator=separator):
+                with tempfile.TemporaryDirectory() as path:
+                    filepath = os.path.join(path, 'file.json')
+                    settings = self.create_settings(json_file=filepath, json_thousands_separator=separator, json_test_case_results=True)
+
+                    gh, gha, req, repo, commit = self.create_mocks(digest=self.base_digest, check_names=[settings.check_name])
+                    publisher = Publisher(settings, gh, gha)                    
+
                     publisher.publish_json(self.publish_data)
                     gha.error.assert_not_called()
 
                     # assert the file
                     with open(filepath, encoding='utf-8') as r:
+                        self.maxDiff = None
```

Comment:
not needed once the tests pass, we can add this when we need to debug a test (I hope this is true for the whole project)

Reply:
It's not, but I can remove the one other place I saw it at as part of this PR as well